v2/internal/runner/inputs.go

```diff
@@ -0,0 +1,77 @@
+package runner
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/projectdiscovery/gologger"
+	"github.com/projectdiscovery/hmap/store/hybrid"
+	"github.com/projectdiscovery/nuclei/v2/pkg/protocols/http/httpclientpool"
+	"github.com/projectdiscovery/retryablehttp-go"
+	"github.com/remeh/sizedwaitgroup"
+)
+
+// initializeTemplatesHTTPInput initializes the http form of input
+// for any loaded http templates if input is in non-standard format.
+func (r *Runner) initializeTemplatesHTTPInput() (*hybrid.HybridMap, error) {
+	hm, err := hybrid.New(hybrid.DefaultDiskOptions)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create temporary input file")
+	}
+
+	httpclient, err := httpclientpool.Get(r.options, &httpclientpool.Configuration{})
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get http client")
+	}
+	gologger.Verbose().Msgf("Started probing of non-http URLs")
+
+	// Probe the non-standard URLs and store them in cache
+	swg := sizedwaitgroup.New(r.options.BulkSize)
+	count := 0
+	r.hmapInputProvider.Scan(func(value string) bool {
+		if strings.Contains(value, "://") {
```

Comment:
What do you think about making the test stricter by checking the prefix `https://` or `http://`, since there are many edge cases like:
- Connection strings (eg. `sql://`, `ftp://`, etc)
- URLs (eg. `host:port?param=https://xxxx`)

Reply:
done